scripts/extractStrings.js

```diff
@@ -0,0 +1,271 @@
+  }
+}
+
+function getLocProjectFilePath(templatePath) {
+  var root = "";
+  if (templatePath.includes(CohortsTemplateFolder)) {
+    root = templatePath.slice(0,templatePath.indexOf(CohortsTemplateFolder));
+  } else {
+    root = templatePath.slice(0,templatePath.indexOf(WorkbookTemplateFolder));
+  }
+  return root.concat("\\src");
+}
+
+/** Write a LocProject.json file needed to point the loc tool to the resjson files + where to output LCL files  */
+function generateLocProjectFile(locItems, templatePath) {
+  const locProjectJson = {
+    "Projects": [
+      {
+        "LanguageSet": "Azure_Languages",
+        "LocItems": locItems
+      }
+    ]
+  };
+  const content = JSON.stringify(locProjectJson, null, "\t");
+  try {
+    const pathToLocProjectFile = getLocProjectFilePath(templatePath);
+    if (!fs.existsSync(pathToLocProjectFile)) {
+      fs.mkdirSync(pathToLocProjectFile, { recursive: true });
+    }
+    fs.writeFileSync(pathToLocProjectFile.concat("\\", LocProjectFileName), content);
+    console.log(">>>>> Generated LocProject.json file: ", pathToLocProjectFile);
+  } catch (e) {
+    console.error("Cannot write LocProject.json file: ", pathToLocProjectFile, "ERROR: ", e);
+  }
+}
+
+/**
+ * 
+ * SCRIPT MAIN
+ * 
+ */
+
+if (!process.argv[2]) { // path to extract strings from 
+  console.log('ERROR: Workbook path not provided. Please provide the path to the workbook folder.');
+  return;
+}
+
+// Verify template path
+const templatePath = process.argv[2];
+const exists = testPath(templatePath);
+if (!exists) {
+  return;
+}
+
+// Valid args, start processing the files.
+console.log("Processing...");
+
+// This is where we will output the resjson artifact
+const RESJSONOutputPath = generateRESJSONOutputPath(templatePath);
+
+const locItems = [];
+
+const files = fs.readdirSync(templatePath);
+
+// Create string output dir
+if (!fs.existsSync(RESJSONOutputPath)) {
+  fs.mkdirSync(RESJSONOutputPath, { recursive: true });
+}
+
+for (var i in files) {
+  const fileName = files[i];
+  // Get the contents of the workbook
+  const isValid = isValidFileType(fileName);
+  if (!isValid) {
+    continue;
+  }
+
+  const filePath = templatePath.concat("\\", fileName);
+  const data = openFile(filePath);
+
+  // parse the workbook for strings
+  var extracted = {};
+  try {
+    getLocalizeableStrings(JSON.parse(data), '', extracted, fileName);
+  } catch (error) {
+    console.log("ERROR: Cannot extract JSON: ", filePath);
+    continue;
+  }
+
+  if (Object.keys(extracted).length > 0) {
+    const LCLOutputPath = templatePath.concat(StringOutputPath);
+    const resjsonFileName = getResJSONFileName(fileName);
+
+    // Add LocProject entry
+    locItems.push({
+      "SourceFile": RESJSONOutputPath.concat("\\", resjsonFileName),
+      "CopyOption": "LangIDOnPath",
+      "OutputPath": LCLOutputPath
+    });
+
+    // Write extracted strings to file
+    writeToFileRESJSON(extracted, fileName, RESJSONOutputPath);
```

Comment:
is there a resource file for each template? the build loc scripts sees to only process on one file